MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -434,6 +440,14 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
 
       // calculate animation time based on displacement
       long animationTime = (long) (velocityXY / 7 / tiltFactor + MapboxConstants.ANIMATION_DURATION_FLING_BASE);
+      if (!uiSettings.isHorizontalScrollGesturesEnabled()) {
+        // determine if angle of fling is valid for performing a vertical fling
+        double angle = Math.abs(Math.toDegrees(Math.atan(offsetX / offsetY)));
+        if (angle > MapboxConstants.ANGLE_THRESHOLD_IGNORE_VERTICAL_FLING) {
```

Comment:
Instead of doing this calculation, could you only check the `velocityY` (and come up with a new threshold) like above?

Reply:
That was the initial implementation, see https://github.com/mapbox/mapbox-gl-native-android/pull/156#discussion_r369763501. IMO this is a cleaner solution as we calculate where the fling is going, just checking the velocityY would result in ignore short vertical flings.

Reply:
I think that's okay, we shouldn't really care how fast horizontally is the fling and currently, it will impact the calculation in 
https://github.com/mapbox/mapbox-gl-native-android/pull/156/files#diff-40fbbd818189e7d34788f0a24144c366R422-R426.

My idea would be to ignore the above check and replace it with one that checks only the `velocityY` (not `offsetY`) if the horizontal scroll is disabled.

Also, the verification in any form has to be moved up, so that we are not dispatching any events if it fails.

Reply:
> I think that's okay

I disagree, we have more certainty about how the gesture has been performed. Doesn't make sense saying with going with a less certain solution. 

> we shouldn't really care how fast horizontally is the fling 

I don't care either, I only care about the angle of the gesture

> and currently, it will impact the calculation in

Please explain how this would impact that calculation

>   Also, the verification in any form has to be moved up, so that we are not dispatching any events if it fails.

sure

Reply:
Tested out your requested changes in `tvn-alternative-approach` and the UX is worse. If you have large fling, even with not much y displacement the values of velocityY sky rocket.

Reply:
Thanks @tobrun. I tested as well and the current setup feels alright :+1: